samples/client.c

```diff
@@ -492,12 +491,48 @@ access_region(int sock, int region, bool is_write, uint64_t offset,
         err(EXIT_FAILURE, "failed to alloc recv_data");
     }
 
-    pthread_mutex_lock(&mutex);
-    ret = tran_sock_msg_iovec(sock, msg_id--, op,
```

Comment:
Isn't it easier to e.g. extend `tran_sock_send_iovec` to accept a `no_reply` option

Reply:
tran_*() is a private interface not API/ABI

Reply:
> That would make more sense but it would be a a pretty big breaking change, does that make a difference?

I don't think it's such a big change TBH, it's straightforward to modify the callers.

> tran_*() is a private interface not API/ABI

We've already been using tran_*() ins the client/server samples, should we remove it entirely from there or is it "OK" to use them as long as we don't require additional functionality (like the `no_reply` argument I suggested)?

Reply:
In that case yes I'll change it.

Reply:
we can do whatever we like with our private code...
Although I would hope we're not using tran_*() in server: our APIs are supposed to cover everything a server needs.

Reply:
on that note can we remove tran_sock.h from server.c?

Reply:
We're not using `train_*()` in `server.c` but yeah we still have the include, should I make another PR to remove that? What am I doing with this one? @jlevon

Reply:
> on that note can we remove tran_sock.h from server.c?

#752